src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -315,6 +324,12 @@ private static InputStreamWithMetadata compressStreamWithGZIP(
 
       countingStream.flush();
 
+      // Normal flow will never hit here. This is only for testing purposes
+      if (isInjectedFileTransferExceptionEnabled()
+          && injectedFileTransferException instanceof NoSuchAlgorithmException) {
```

Comment:
why can't we make new method like void checkForInjectedTransferExceptionEnabled() throws NoSuchAlgorithmException
{
if (isInjectedFileTransferExceptionEnabled()
          && injectedFileTransferException instanceof NoSuchAlgorithmException) {
        throw (NoSuchAlgorithmException) SnowflakeFileTransferAgent.injectedFileTransferException;
      }
}

then calll this method everywhere.